### DISCUSSION TITLE
Failed to open TCP connection to rubygems.org:443

### DISCUSSION THREAD

**Discussion by Kolkman** [mastodon/mastodon#26127] [Q&A]
When I try to build the docker image the build process terminates with: 
```
#0 250.0 Retrying fetcher due to error (4/4): Bundler::HTTPError Could not fetch specs from https://rubygems.org/ due to underlying error <Net::OpenTimeout: Failed to open TCP connection to rubygems.org:443 (execution expired) (https://rubygems.org/specs.4.8.gz)>
```
I noticed that disabling IPv6 in `/etc/docker/daemon.json` leads to a successful build. However, I rather not restart docker to disable IPv6 and reenable it after the Mastodon build.

I have tried to use a IPv6 only build context:

```
docker network create --ipv6=false -driver=bridge   disable_ipv6
docker buildx use $(docker buildx create --driver-opt network=disable_ipv6)
```

but to no avail.

Does anybody have any tips?

**Comment by Kolkman** [comment]
Update: Turns out that the `buildx` context trick above does work, at least for now.

(This has been bugging me for many version updates and I hope that this solution is a consistent one.)

**Comment by renchap** [comment]
This points to an issue with your Docker (or server) setup, which does not have IPv6 working.

This is not Mastodon-related.

**Comment by Kolkman** [comment]
Indeed it is not Mastodon-related (but for NOOBs like me the Mastodon build is where we might run into the issue).

Although I am puzzled about what is wrong exactly - I have v6 networking enabled and my Treafik container uses it, but it seems like the network used by the build process is not v6 enabled.  

Anyway, for the benefit of others who might be experiencing this, it turns out that there is an even simpler way to work around this issue:

```
docker buildx build --network host .
```
